shrinidhihegde3/Snippify-Hub
# Updated code for Vigenère Cipher issue no #41

Fixed: #41 
## Description

This pull request includes the following updates:

1. **Vigenère Cipher Component**:
    - Created the main `VigenereCipher.tsx` file, implementing a fully functional Vigenère Cipher encryption and decryption feature.
    - Allows users to input a message and a key to either encrypt or decrypt the text.
    - Implemented input validation to ensure both text and key are provided before processing.
    - Used React's `useState` hook to manage text, key, and output states.
    - Added an interactive UI with Tailwind CSS, including buttons to toggle between encryption and decryption modes.

2. **CodeSnippet Component**:
    - Updated the `CodeSnippet.tsx` file to handle code block rendering for various languages with copy-to-clipboard functionality.
    - Syntax highlighting using `react-syntax-highlighter` with the **Darcula** theme.
    - Implemented a copy-to-clipboard feature with a success notification.
    - Responsive and styled using Tailwind CSS.

## Changes Made

- Created the main `VigenereCipher.tsx` file to include the cipher logic and the UI.
- Updated the `CodeSnippet.tsx` to enhance code rendering and interaction.
- Improved styling across components for better UI/UX using Tailwind CSS.

## How to Test

1. Enter text and a keyword to either encrypt or decrypt.
2. Toggle between encryption and decryption modes using the buttons.
3. Submit to view the result in the output field.
4. Check the Python, C++, and Java code snippets for syntax highlighting and test the copy-to-clipboard functionality.

## Screenshots 
![Screenshot 2024-10-20 120829](https://github.com/user-attachments/assets/4e6e421e-ab31-4ae5-b2d9-4dc2a32bd5c6)
![Screenshot 2024-10-20 120852](https://github.com/user-attachments/assets/798ccaca-9fa8-418f-9595-bac704b8e140)


## Checklist

- [x] Tested the Vigenère Cipher functionality for both encryption and decryption.
- [x] Verified that the `CodeSnippet` component displays properly with syntax highlighting and copy functionality.
- [x] Ensured that all UI components are responsive across different screen sizes.
- [x] Code adheres to project guidelines and coding standards.
